github/spec/features/validate_repos_spec.rb

```diff
@@ -0,0 +1,88 @@
+require_relative '../spec_helper'
+require_relative '../../lib/configure_repos'
+require_relative '../../lib/fetch_repos'
+require_relative '../../lib/validate_repos'
+
+require 'base64'
+require 'yaml'
+require "octokit"
+require "json"
+require "open-uri"
+
+RSpec.describe ValidateRepos do
+
+  it "should ignore any repos that exist in repos.json AND are tagged govuk in GitHub" do
+    repos = [{
+      "app_name": "this-is-a-govuk-repo",
+    }]
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-govuk-repo")
+    octokit_mock = double("Octokit::Client", org_repos: [repo_mock])
+
+    stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
+      to_return(status: 200, body: repos.to_json, headers: {})
+
+    expect { ValidateRepos.new(octokit_mock).verify_repo_tags }.to output("\e[0;32;40mUntagged govuk repos: No mismatches found.\e[0m\n\n\e[0;32;40mFalsely tagged govuk repos: No mismatches found.\e[0m\n").to_stdout
+
+  end
+
+  it "should alert if finds an untagged repo in repos.json" do
+    
+    repos = [
+      { "app_name": "this-is-a-govuk-repo", },
+      { "app_name": "this-govuk-repo-is-not-tagged!", }
+  ]
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-govuk-repo")
+    octokit_mock = double("Octokit::Client", org_repos: [repo_mock])
+
+    stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
+      to_return(status: 200, body: repos.to_json, headers: {})
```

Comment:
This looks identical to the test setup above it, except that `repos` contains two elements instead of one.

I think you could move lines 39 to 43 into a `before` method (see [example](https://github.com/alphagov/content-publisher/blob/278819c03881c7803bfa9516494a373b61aa319e/spec/jobs/whitehall_document_import_job_spec.rb#L16-L23)) which runs before every test, meaning you don't need to duplicate this anywhere.

Then this test would literally be:

```ruby
repos = [
  # ...
]

stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
  to_return(status: 200, body: repos.to_json, headers: {})

expect { # ...
```

You'll probably want to make a helper method at the bottom of this file too actually, something like:

```ruby
def stub_repos_json(repos)
  stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
    to_return(status: 200, body: repos.to_json, headers: {})
end
``` 

So that the test would just be:


```ruby
stub_repos_json([
  # ...
])

expect { # ...
```